portal/main.py

```diff
@@ -44,3 +44,69 @@ def my_courses():
     cur.close()
 
     return render_template("layouts/home.html", courses=courses)
+
+
+@bp.route("/course/<int:course_id>/session/<int:session_id>/assignments/<int:assignment_id>/grades")
+@teacher_required
+@login_required
```

Comment:
Since the above teacher required means a teacher is already logged in, there isn't a need for the login required, unless it is before the teacher required.